modules/product/src/Form/AddToCartForm.php

```diff
@@ -150,9 +149,29 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     if (!$cart) {
       $cart = $this->cartProvider->createCart('default', $store);
     }
-    $quantity = $form_state->getValue('quantity');
+
     $combine = $form['#settings']['combine'];
-    $this->cartManager->addEntity($cart, $variation, $quantity, $combine);
+
+    /** @var \Drupal\commerce_product\Entity\Product $product */
+    $product = $form['#product'];
+    $line_item_form_display = $this->getLineItemFormDisplay($product);
+
+    $line_item = LineItem::create([
```

Comment:
This needs to be $this->cartManager->createLineItem().
LineItem:: is generally only allowed in tests, real classes must use the storage, but in this case we actually need the cartManager method.